core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -614,8 +612,10 @@ public long getWorkerId(WorkerNetAddress workerNetAddress) {
     }
 
     // Generate a new worker id.
-    long workerId = mNextWorkerId.getAndIncrement();
-    mWorkers.add(new MasterWorkerInfo(workerId, workerNetAddress));
+    long workerId = IdUtils.getRandomNonNegativeLong();
+    while (!mWorkers.add(new MasterWorkerInfo(workerId, workerNetAddress))) {
```

Comment:
I don't think this will work because two workers might get the same ID but have different workerNetAddresses, so their MasterWorkerInfos will be unequal.